### PR TITLE
feat: add reason for discards [PIPE-1473]

### PR DESCRIPTION
**Pull request by BonapartePC**
# Description

add reason for discards

## Linear Ticket

https://linear.app/rudderstack/issue/PIPE-1473/add-reason-for-discards

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by achettyiitr — APPROVED** [review]

File: warehouse/slave/worker.go [hunk 1]
```diff
@@ -31,11 +31,7 @@ import (
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
-var (
-	errIncompatibleSchemaConversion = errors.New("incompatible schema conversion")
-	errSchemaConversionNotSupported = errors.New("schema conversion not supported")
-	json                            = jsoniter.ConfigCompatibleWithStandardLibrary
-)
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
```
Comment (cisse21) [inline warehouse/slave/worker.go:34]:
```suggestion
var (
json = jsoniter.ConfigCompatibleWithStandardLibrary
errIncompatibleSchemaConversion = "incompatible schema conversion from %v to %v"
errSchemaConversionNotSupported = "incompatible schema conversion from %v to %v"
)
```

File: warehouse/slave/worker.go [hunk 2]
```diff
@@ -534,14 +539,14 @@ func handleSchemaChange(existingDataType, currentDataType model.SchemaType, valu
 	} else if (currentDataType == model.IntDataType || currentDataType == model.BigIntDataType) && existingDataType == model.FloatDataType {
 		intVal, ok := value.(int)
 		if !ok {
-			err = errIncompatibleSchemaConversion
+			err = fmt.Errorf("incompatible schema conversion from %v to %v", existingDataType, currentDataType)
```
Comment (cisse21) [inline warehouse/slave/worker.go:542]:
```suggestion
			err = fmt.Errorf(errIncompatibleSchemaConversion, existingDataType, currentDataType)
```

File: warehouse/slave/worker.go [hunk 3]
```diff
@@ -534,14 +539,14 @@ func handleSchemaChange(existingDataType, currentDataType model.SchemaType, valu
...
 		} else {
 			newColumnVal = float64(intVal)
 		}
 	} else if currentDataType == model.FloatDataType && (existingDataType == model.IntDataType || existingDataType == model.BigIntDataType) {
 		floatVal, ok := value.(float64)
 		if !ok {
-			err = errIncompatibleSchemaConversion
+			err = fmt.Errorf("incompatible schema conversion from %v to %v", existingDataType, currentDataType)
```
Comment (cisse21) [inline warehouse/slave/worker.go:549]:
```suggestion
			err = fmt.Errorf(errIncompatibleSchemaConversion, existingDataType, currentDataType)
```

File: warehouse/slave/worker.go [hunk 4]
```diff
@@ -555,7 +560,7 @@ func handleSchemaChange(existingDataType, currentDataType model.SchemaType, valu
 			newColumnVal = fmt.Sprintf(`"%v"`, value)
 		}
 	} else {
-		err = errSchemaConversionNotSupported
+		err = fmt.Errorf("incompatible schema conversion from %v to %v", existingDataType, currentDataType)
```
Comment (cisse21) [inline warehouse/slave/worker.go:563]:
```suggestion
		err = fmt.Errorf(errSchemaConversionNotSupported, existingDataType, currentDataType)
```

**Review by cisse21 — APPROVED** [review]
Approved with non blocking comments

**Pull request merged**